sklearn/metrics/cluster/supervised.py

```diff
@@ -188,33 +204,49 @@ def adjusted_rand_score(labels_true, labels_pred, max_n_classes=5000):
     adjusted_mutual_info_score: Adjusted Mutual Information
 
     """
-    labels_true, labels_pred = check_clusterings(labels_true, labels_pred)
-    n_samples = labels_true.shape[0]
-    classes = np.unique(labels_true)
-    clusters = np.unique(labels_pred)
+    if contingency is None:
+        labels_true, labels_pred = check_clusterings(labels_true, labels_pred)
+        n_samples = labels_true.shape[0]
+        n_classes = np.unique(labels_true).shape[0]
+        n_clusters = np.unique(labels_pred).shape[0]
+    elif isinstance(contingency, _data_matrix):  # scipy.sparse.data._data_matrix
+        n_samples = contingency.nnz
+        n_classes, n_clusters = contingency.shape
+    else:
+        raise ValueError("'contingency' must be a sparse matrix or None")
```

Comment:
I don't understand. `sklearn.utils.check_array` ? I didn't change the logic of this at all